cpu/sam0_common/periph/uart.c

```diff
@@ -57,6 +57,9 @@ int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
         return UART_NODEV;
     }
 
+    /* must disable here first to ensure idempotency */
+    uart_poweroff(uart);
```

Comment:
hmm, I think the cleaner way would be to simply call `dev(uart)->CTRLA.reg &= ~(SERCOM_USART_CTRLA_ENABLE);` here. In theory that should do the trick on both?!

Reply:
Yes it should work as uart_poweroff disabled sercom + peripheral clock but the peripheral clock is re-enable just a few line after.
In fact, the issues with SAML21 family are the write-protected registers when CTRLA Enable bit is set. We cannot set this bit within uart_poweron because normally this function is called early in <periph>_init. Most of the parameters of the peripheral will then be locked. uart_poweron shoud only enable clock configuration but CTRLA enable bit must be set at the very end of the <periph>_init function. (for SAML21 family only, SAMD21 doesn't seem to have these write-protected registers).

Reply:
Well we would need a test on SAML21 to see if it doesn't break anything, but I agree with @haukepetersen that only disabling the peripheral is cleaner, since anyways the clock is re-activated some instructions afterwards.

Reply:
changed to just unset the enable bit. On samr21-xpro, this still ifxes the idempotency issue.